src/nvim/tui/input.c

```diff
@@ -30,6 +44,61 @@ void term_input_init(TermInput *input, Loop *loop)
   uv_cond_init(&input->key_buffer_cond);
 
   const char *term = os_getenv("TERM");
+#ifdef WIN32
+  bool conemu_ansi = false;
+  bool winpty = false;
+  bool vtp = false;
+  const char *env = os_getenv("VIM_TERMINAL");
+  if (env) {
+    winpty = true;
+  }
+  // If we change to set environment variable in terminal of nvim,
+  // add condition here
+
+  if (!winpty) {
+# ifdef NVIM_UV_HAS_SET_VTERM_STATE
+    env = os_getenv("ConEmuANSI");
+    if (env && !STRCMP(env, "ON")) {
+      conemu_ansi = true;
+    }
+# endif
+
+    HANDLE handle = (HANDLE)_get_osfhandle(1);
+    DWORD dwMode;
+    if (handle != INVALID_HANDLE_VALUE && GetConsoleMode(handle, &dwMode)) {
+      dwMode |= ENABLE_VIRTUAL_TERMINAL_PROCESSING;
+      if (SetConsoleMode(handle, dwMode)) {
+        vtp = true;
+      }
+    }
+  } else {
+    // If it is running under winpty ignore the TERM environment variable and
+    // force it to be win32con.
+    term = "win32con";
+  }
+
+  if (term == NULL) {
+    if (vtp) {
+      term = "vtpcon";
+    } else if (conemu_ansi) {
+      term = "conemu";
+    } else {
+      term = "win32con";
+    }
+  }
+
+  os_setenv("TERM", term, 1);
+# ifdef NVIM_UV_HAS_SET_VTERM_STATE
+  if (conemu_ansi) {
+    uv_set_vterm_state(UV_SUPPORTED);
+  } else if (winpty) {
+    uv_set_vterm_state(UV_UNSUPPORTED);
+  }
+# endif
+# ifdef NVIM_UNIBI_HAS_SET_FALLBACK
+  unibi_set_fallback(unibi_fallback);
```

Comment:
Comment would be helpful, e.g.:

    // libtermkey uses unibi_from_term internally